ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh

```diff
@@ -115,3 +95,32 @@ cat > packet-setup.yaml <<-EOF
 EOF
 
 ansible-playbook packet-setup.yaml -e "packet_hostname=ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID}"  |& gawk '{ print strftime("%Y-%m-%d %H:%M:%S"), $0; fflush(); }'
+
+DEVICEID=$(jq -r .devices[0].id < ${SHARED_DIR}/hosts.json)
+
+function refresh_device_info(){
+    curl -H "X-Auth-Token: $(cat ${CLUSTER_PROFILE_DIR}/packet-auth-token)"  "https://api.equinix.com/metal/v1/devices/$DEVICEID" > /tmp/device.json
+    STATE=$(jq -r .state < /tmp/device.json)
+    IP=$(jq -r .ip_addresses[0].address < /tmp/device.json)
+}
```

Comment:
Implemention is fine, but just for the sake of consistency my preference would be to have them as Ansible task (I think the [uri](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/uri_module.html) module could be used for API calls, and also [shell](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/shell_module.html) could be useful)

Reply:
I intentionally stayed away from ansible as we'd be more likely to see the raw errors with curl (i.e. ansible is hiding what actually happened from us)

Reply:
We could add the `{{ hosts }}` to the slack reporting message also to fetch it

Reply:
I've added the contents of the json to the slack message but I'm also interested in seeing HTTP status codes so stuck to using curl for the moment.